evm/account.cpp

```diff
@@ -57,15 +57,19 @@ namespace evm
   {
     j["address"] = a.address;
     j["balance"] = a.balance;
-    j["nonce"] = a.nonce;
-    j["code"] = a.code;
+    j["nonce"] = std::to_string(a.nonce);
```

Comment:
This will produce a decimal string, but should produce a hex-string.

Reply:
Yes, you are right. 

First, I thought to use:
`j["nonce"] = to_hex_string(to_bytes(std::to_string(a.nonce)));` 
This does the job, but there is an extra conversion from `uint64` to `string`; 
We could avoid making that extra conversion using a specialized version of `to_bytes`:
```
  std::vector<uint8_t> to_bytes(uint64_t v);                                                                                             
  std::vector<uint8_t> to_bytes(const std::string& s);  
```
but it turned out to be not very convenient solution, because it introduces type ambiguities,
usually in assign_j calls, like:
`     assign_j(a.code, to_bytes(j["code"])); `
Here, it is no longer obvious which to_bytes should be used. It can be eliminated by introducing explicit type conversion, e.g. `std::string(j["code|"])`, but in this case a lot of code must be updated.

We can also introduce a function with another name, and use it alongside with `to_bytes`.
What do you think?

Reply:
I think rather than converting `uint64_t` to a byte vector, we should just convert it directly to a hex string. Either with a specialisation of `to_hex_string`, or if that introduces type ambiguities then add a new function with a different name (`uint_to_hex_string`).

Reply:
Done.